### PR TITLE
Fundies

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-14580

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/fundies/sdk/dotnet/fundamentals/mongodb-realm-cloud/

### CODE REVIEW

File: source/sdk/dotnet/fundamentals.txt [hunk 1]
```diff
@@ -6,6 +6,7 @@ Realm Fundamentals - .NET SDK
    :titlesonly:
    :hidden:
 
+   MongoDB Realm </sdk/dotnet/fundamentals/mongodb-realm-cloud>
```
Comment (cbush) [inline source/sdk/dotnet/fundamentals.txt:9]: Be sure to put this at the bottom of the list. Top is for local-only (Realm Database, non-Sync, free, etc.) use case.

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 2]
```diff
@@ -1,13 +1,75 @@
...
+   Each SDK section in the Realm documentation includes code examples for 
+   accomplishing the functionality discussed here. Find the platform / language 
+   you are working with and look under **Usage Examples**.
+
+The App Client
+----------------
+The fundamental building block in the SDK is the ``App`` object. It is from this 
+object that all other functionality is provided. The ``App`` object is 
+:ref:`initialized <dotnet-init-appclient>` with the Realm app ID, which you can 
+:ref:`find your Realm app ID <find-your-app-id>` in the Realm UI.
```
Comment (cbush) [inline source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:36]:
```suggestion
:ref:`find in the Realm UI <find-your-app-id>`.
```

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 3]
```diff
@@ -1,13 +1,75 @@
...
+
+.. note:: Code Examples
+
+   Each SDK section in the Realm documentation includes code examples for 
+   accomplishing the functionality discussed here. Find the platform / language 
+   you are working with and look under **Usage Examples**.
+
+The App Client
+----------------
+The fundamental building block in the SDK is the ``App`` object. It is from this
```
Comment (cbush) [inline source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:33]: This is only true for the MongoDB Realm stuff, but not for the database. That is, you don't need to log in or do anything with the app if you aren't using Sync, Functions, Remote MongoDB, or auth. So I would reword this to be less about "the SDK" as a whole and "all functionality" and more about interacting with MongoDB Realm and MongoDB Realm functionality.

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 4]
```diff
@@ -1,13 +1,75 @@
...
+--------------
+One of the most challenging aspects of client development is implementing a 
+robust and secure authentication system. With the {+service-short+} SDKs, however, you 
+can use any of the {+service-short+} authentication providers with very minimal backend 
+setup or client-side code required. With the authentication APIs, you can 
+implement the following functionality:
+
+- Creation of new user accounts
+- User log-in and log-out
+- Multiple concurrent users
```
Comment (cbush) [inline source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:48]: Can we clarify this? When we say concurrent, do we mean multiple users logged in at the same time or do we mean allowing you to have different logins/accounts saved on a device?

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 5]
```diff
@@ -1,13 +1,75 @@
...
+- Linking user accounts from different providers
+- Providing custom user data
+
+For more information, see :ref:`Authenticate Users <dotnet-authenticate>`.
+
+Calling Functions
+-----------------
+Using the {+service-short+} SDKs, you can :ref:`create server-side functions <create-a-function>`,
```
Comment (cbush) [inline source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:56]: I don't think you meant _create_ here. But we can call them which is pretty cool too.

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 6]
```diff
@@ -1,13 +1,75 @@
...
+- Linking user accounts from different providers
+- Providing custom user data
+
+For more information, see :ref:`Authenticate Users <dotnet-authenticate>`.
+
+Calling Functions
+-----------------
+Using the {+service-short+} SDKs, you can :ref:`create server-side functions <create-a-function>`, 
+which enables you to call the same functionality from multiple client types 
+without have to reproduce the logic. Calling functions from your client code is
```
Comment (cbush) [inline source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:58]: Typo
```suggestion
without having to reproduce the logic. Calling functions from your client code is 
```

Comment (cbush) [inline source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:58]: This is a good point. What about calling into the "backend as a service" functionality that you have defined? In particular, you can use functions to have server-controlled permissions management.

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 7]
```diff
@@ -1,13 +1,75 @@
...
+which enables you to call the same functionality from multiple client types 
+without have to reproduce the logic. Calling functions from your client code is 
+then reduced to only a few lines of code. 
+
+For more information on calling functions, see :ref:`Call a Function <dotnet-call-a-function>`.
+
+Accessing MongoDB
+-----------------
+While many app clients don't require functionality beyond that provided by 
+{+service-short+}, the Realm SDKs include the APIs for accessing a MongoDB
```
Comment (cbush) [inline source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:66]: I'm not sure I get the intention here when you say "most apps don't require...". Is it to appease the sense of feature overload? I'm reading this as a bit of a downplay on the MongoDB Remote Access feature.

**Review by cbush — COMMENTED** [review]
Great first pass! I think this covers all of the intended information. Leaving some comments for now and I'd like to hear your thoughts.

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt (hunk 3 above)
Comment (cbush) [reply source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:33]: Still looking for an update or something here.

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 8]
```diff
@@ -1,13 +1,77 @@
...
+setup or client-side code required. With the authentication APIs, you can 
+implement the following functionality:
+
+- Creation of new user accounts
+- User log-in and log-out
+- Multiple users logged on at the same time on the same device
+- Linking user accounts from different providers
+- Providing custom user data
+
+For more information, see :ref:`Authenticate Users <dotnet-authenticate>`.
```
Comment (cbush) [inline source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:52]: It might be nice to actually link out to the individual usage examples/advanced guides pages about each of the above.

**Review by cbush — COMMENTED** [review]
Looks great. Still wondering about the "fundamental building block" part.

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 9]
```diff
@@ -1,13 +1,77 @@
...
+call backend :ref:`functions <functions>`, and directly access a 
+:ref:`MongoDB database <mongodb-service>`. 
+
+.. note:: Code Examples
+
+   Each SDK section in the Realm documentation includes code examples for 
+   accomplishing the functionality discussed here. Find the platform / language 
+   you are working with and look under **Usage Examples**.
+
+The App Client
```
Comment (nlarew) [inline source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:31]:
```suggestion
The Realm App Client
```
nit: For consistency and clarity I think it's a good idea to always specify that it's a _Realm_ app client. This is particularly true given its code type is usually some variant e.g. `RealmApp` or `Realm.App`

**Review by nlarew — CHANGES_REQUESTED** [review]
A few changes to contextualize to .NET + some broader thoughts

**Comment by nlarew** [comment]
Also looks like some extra commits snuck in here. Let's rebase on top of `new-ia`

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt (hunk 8 above)
Comment (MongoCaleb) [reply source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:52]: Removed the "fundamental building block" stuff (sorry I missed that!).

As for the links, the problem is that the first 2 items in the list really depend on which provider. I can link to the main page, which I do below the list already. I've checked in the list now with the others linked. LMK your thoughts.

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 10]
```diff
@@ -1,13 +1,77 @@
-.. _dotnet-mongodb-realm-cloud:
+.. _mongodb-realm-cloud:
 
-==============================
-MongoDB Realm Cloud - .NET SDK
-==============================
+=======================================
+MongoDB Realm Backend Applications
+=======================================
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:5]: Other than esthetic, it's not a problem to have them longer than the text. But fixed.

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 11]
```diff
@@ -1,13 +1,77 @@
...
+backend. The SDK provides the {+service+} functionality needed to 
+authenticate users with any of the built-in :ref:`Authentication providers <authentication-providers>`, 
+call backend :ref:`functions <functions>`, and directly access a 
+:ref:`MongoDB database <mongodb-service>`. 
+
+.. note:: Code Examples
+
+   Each SDK section in the Realm documentation includes code examples for 
+   accomplishing the functionality discussed here. Find the platform / language 
+   you are working with and look under **Usage Examples**.
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:29]: I was intentionally writing it to be generic enough that we could copy-paste without too much replacement, but I see now that's kinda silly....

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt (hunk 8 above)
Comment (MongoCaleb) [reply source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:53]: @nlarew @cbush now that I'm seeing the staging, I really don't like having some items linked and others not.

File: source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt [hunk 12]
```diff
@@ -1,13 +1,77 @@
...
+
+Calling Functions
+-----------------
+Using the {+service+} SDKs, you can call server-side functions, 
+which enables you to reuse the same backend functionality from multiple client types 
+without having to reproduce the logic client-side. Calling functions from your client code is 
+then reduced to only a few lines of code. In addition, server-side functions can 
+run in the context of a system user, giving them full access to MongoDB CRUD and 
+Aggregation APIs regardless of any rules, roles, or permissions assigned to a 
+collection.
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/sdk/dotnet/fundamentals/mongodb-realm-cloud.txt:63]: @nlarew LMK what you think of the change.

**Review by cbush — COMMENTED** [review]
Page looks great, but something doesn't look right with this git history. I would reset to upstream/new-ia and cherry-pick all of the relevant commits, then force push back up.
